### PR TITLE
Pixiv

### PR DESCRIPTION
**Pull request by ombe1229**

### CODE REVIEW

File: utils/pixiv.py [hunk 1]
```diff
@@ -48,38 +47,44 @@ async def request(method, endpoint, **kwargs):
...
+            return discord.Embed(title="해당 작품은 삭제되었거나 존재하지 않는 작품 ID입니다.")
+        return PixivIllustModel(
+            resp["body"]["id"],
+            resp["body"]["title"],
+            resp["body"]["urls"]["original"],
+            resp["body"]["userName"],
+        )
 
 
 class PixivExt(PixivRequester):
```
Comment (SaidBySolo) [inline utils/pixiv.py:85]:
```suggestion
 class PixivResolver(PixivRequester):
```

File: utils/pixiv.py [hunk 2]
```diff
@@ -1,42 +1,41 @@
...
+class PixivInfoModel:
+    def __init__(self, bookmark, comment, id, like, title, username, date, view):
         self.bookmark = bookmark
         self.comment = comment
-        self.id = _id
+        self.id = id
         self.like = like
         self.title = title
         self.username = username
-        self.uploadDate = uploadDate
```
Comment (SaidBySolo) [inline utils/pixiv.py:19]:
```suggestion
        self.upload_date = upload_date
```

File: utils/pixiv.py [hunk 3]
```diff
@@ -1,42 +1,41 @@
...
 from bs4 import BeautifulSoup
+import re
+import time
+from datetime import datetime
 
 
-class PixivModel:
-    def __init__(self, bookmark, comment, _id, like, title, username, uploadDate, view):
+class PixivInfoModel:
+    def __init__(self, bookmark, comment, id, like, title, username, date, view):
```
Comment (SaidBySolo) [inline utils/pixiv.py]:
```suggestion
    def __init__(self, bookmark, comment, id, like, title, username, upload_date, view):
```

**Review by SaidBySolo — CHANGES_REQUESTED** [review]
Use utils.request.Requst

File: utils/pixiv.py [hunk 4]
```diff
@@ -48,41 +47,47 @@ async def request(method, endpoint, **kwargs):
...
+            return discord.Embed(title="해당 작품은 삭제되었거나 존재하지 않는 작품 ID입니다.")
+        return PixivInfoModel(
+            resp["body"]["bookmarkCount"],
+            resp["body"]["illustComment"],
+            resp["body"]["id"],
+            resp["body"]["likeCount"],
+            resp["body"]["title"],
+            resp["body"]["userName"],
+            resp["body"]["uploadDate"],
+            resp["body"]["viewCount"],
```
Comment (SaidBySolo) [inline utils/pixiv.py]: resp = resp["body"]

resp["bookmarkCount"]

File: utils/pixiv.py [hunk 5]
```diff
@@ -48,41 +47,47 @@ async def request(method, endpoint, **kwargs):
...
+
+    async def get_illust(self, index: int):
+        resp = await self.request("GET", f"/ajax/illust/{index}")
+        if not resp:
+            return discord.Embed(title="해당 작품은 삭제되었거나 존재하지 않는 작품 ID입니다.")
+        return PixivIllustModel(
+            resp["body"]["id"],
+            resp["body"]["title"],
+            resp["body"]["urls"]["original"],
+            resp["body"]["userName"],
```
Comment (SaidBySolo) [inline utils/pixiv.py]: same

**Review by SaidBySolo — CHANGES_REQUESTED** [review]

**Review by SaidBySolo — CHANGES_REQUESTED** [review]
Why not use utils.request.Requst

**Comment by ombe1229** [comment]
Forgot

File: utils/pixiv.py [hunk 6]
```diff
@@ -179,8 +171,8 @@ async def ranking_embed(self, mode):
             await asyncio.gather(
                 *list(
                     map(
-                        self.make_ranking_illust_embed,
-                        PixivRankingModel.generator_pixiv_ranking_info(
+                        self.make_illust_embed,
+                        PixivIllustModel.generate_pixiv_ranking_info(
                             ranking["contents"]
                         ),
                     )
```
Comment (SaidBySolo) [inline utils/pixiv.py:178]: Can you tell me the time difference between gather and wait?

**Review by SaidBySolo — APPROVED** [review]

**Comment by SaidBySolo** [comment]
Merge into #135

**Pull request merged**